File: demo/src/main/java/com/bytedance/scenedemo/navigation/forresult/SceneResultScene_0.java
package com.bytedance.scenedemo.navigation.forresult;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.bytedance.scene.Scene;
import com.bytedance.scene.interfaces.PushOptions;
import com.bytedance.scene.interfaces.PushResultCallback;
import com.bytedance.scenedemo.R;
import com.bytedance.scenedemo.utility.ColorUtil;

/**
 * Created by JiangQi on 8/3/18.
 */
public class SceneResultScene_0 extends Scene {

    @NonNull
    @Override
    public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return (ViewGroup) inflater.inflate(R.layout.basic_layout, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        getView().setBackgroundColor(ColorUtil.getMaterialColor(getResources(), 1));

        TextView name = getView().findViewById(R.id.name);
        name.setText(getNavigationScene().getStackHistory());

        Button btn = getView().findViewById(R.id.btn);
        btn.setText(getString(R.string.nav_result_scene_to_scene_btn_0));
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getNavigationScene().push(SceneResultScene_1.class, null, new PushOptions.Builder().setPushResultCallback(new PushResultCallback() {
                    @Override
                    public void onResult(@Nullable Object result) {
                        if (result != null) {
                            Toast.makeText(getActivity(), result.toString(), Toast.LENGTH_SHORT).show();
                        }
                    }
                }).build());
            }
        });

        Button btn2 = getView().findViewById(R.id.btn2);
        btn2.setVisibility(View.VISIBLE);
        btn2.setText(getString(R.string.nav_result_scene_to_scene_btn_1));
        btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getNavigationScene().push(SceneResultScene_0.class, null, new PushOptions.Builder().clearTask().setPushResultCallback(new PushResultCallback() {
                    @Override
                    public void onResult(@Nullable Object result) {
                        if (result != null) {
                            Toast.makeText(getActivity(), result.toString(), Toast.LENGTH_SHORT).show();
                        }
                    }
                }).build());
            }
        });
    }
}

File: demo/src/main/java/com/bytedance/scenedemo/navigation/forresult/SceneResultScene_0.kt
package com.bytedance.scenedemo.navigation.forresult

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import com.bytedance.scene.Scene
import com.bytedance.scene.interfaces.PushOptions
import com.bytedance.scenedemo.R
import com.bytedance.scenedemo.utility.ColorUtil

/**
 * Created by JiangQi on 8/3/18.
 */
class SceneResultScene_0 : Scene() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        return inflater.inflate(R.layout.basic_layout, container, false) as ViewGroup
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view.setBackgroundColor(ColorUtil.getMaterialColor(resources, 1))

        val name = view.findViewById<TextView>(R.id.name)
        name.text = navigationScene!!.stackHistory

        val btn = view.findViewById<Button>(R.id.btn)
        btn.text = getString(R.string.nav_result_scene_to_scene_btn_0)
        btn.setOnClickListener {
            requireNavigationScene().push(SceneResultScene_1::class.java, null,
                    PushOptions.Builder().setPushResultCallback { result ->
                        if (result != null) {
                            Toast.makeText(activity, result.toString(), Toast.LENGTH_SHORT).show()
                        }
                    }.build())
        }

        val btn2 = view.findViewById<Button>(R.id.btn2)
        btn2.visibility = View.VISIBLE
        btn2.text = getString(R.string.nav_result_scene_to_scene_btn_1)
        btn2.setOnClickListener {
            requireNavigationScene().push(SceneResultScene_0::class.java, null,
                    PushOptions.Builder().clearTask().setPushResultCallback { result ->
                        if (result != null) {
                            Toast.makeText(activity, result.toString(), Toast.LENGTH_SHORT).show()
                        }
                    }.build())
        }
    }
}
